File: manji-standard-server-ts-hono/src/usecase/user-usecase.ts
import type { User } from "../domain/entity/user.gen.js";
import type { UserService } from "../domain/service/user-service.js";
import type {
  CreateUserInput,
  GetUserInput,
  UserUsecase,
} from "./user-usecase-interface.gen.js";

export class UserUsecaseImpl implements UserUsecase {
  constructor(private readonly userService: UserService) {}

  async createUser(input: CreateUserInput): Promise<User> {
    return this.userService.create(input.email, input.name);
  }

  async getUser(input: GetUserInput): Promise<User | null> {
    return this.userService.getById(input.id);
  }
}

File: manji-standard-server-go/pkg/usecase/user_usecase.go
package usecase

import (
	"context"

	"github.com/example/manji-standard-server-go/pkg/domain/entity"
	"github.com/example/manji-standard-server-go/pkg/domain/service"
)

// UserUsecaseImpl は生成された UserUsecase interface の実装。
type UserUsecaseImpl struct {
	userService *service.UserService
}

var _ UserUsecase = (*UserUsecaseImpl)(nil)

func NewUserUsecase(userService *service.UserService) *UserUsecaseImpl {
	return &UserUsecaseImpl{userService: userService}
}

func (u *UserUsecaseImpl) CreateUser(ctx context.Context, input CreateUserInput) (*entity.User, error) {
	return u.userService.Create(ctx, input.Email, input.Name)
}

func (u *UserUsecaseImpl) GetUser(ctx context.Context, input GetUserInput) (*entity.User, error) {
	return u.userService.GetByID(ctx, input.ID)
}
